clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1667,6 +1667,9 @@ private void updateLastSeenEpochIfNewer(TopicPartition topicPartition, OffsetAnd
     public boolean updateAssignmentMetadataIfNeeded(Timer timer) {
         maybeThrowFencedInstanceException();
         maybeInvokeCommitCallbacks();
+        if (subscriptions.hasPatternSubscription()) {
+            updatePatternSubscription(metadata.fetch());
+        }
```

Comment:
Just for consistency, what about we encapsulate this in something like `maybeUpdateSubscriptionMetadata`? It would align nicely with the above funcs (and also that's how the similar functionality is named in the legacy coordinator so would be helpful to understand how that piece of logic translates into the new consumer)

Reply:
Good call, @lianetm.

Reply:
@lianetm, I understand. Will fix this part